include/mimalloc-internal.h

```diff
@@ -730,7 +734,11 @@ static inline void mi_tls_slot_set(size_t slot, void* value) mi_attr_noexcept {
   tcb[slot] = value;
 #elif defined(__aarch64__)
   void** tcb; UNUSED(ofs);
+#if defined(__MACH__)
+  __asm__ volatile ("mrs %0, tpidrro_el0" : "=r" (tcb));
```

Comment:
This seems unlikely to end well, unless macOS ends up emulating the instruction in order to make it a writable register (which I highly doubt, otherwise surely it'd use `tpidr_el0` instead?).